src/runtime/web/index.js

```diff
@@ -40,9 +40,14 @@ function init() {
   this.views.push(
     document.l10n = new View(this, document));
 
+  var prevLangs = [];
+  if (document.documentElement.hasAttribute('lang')) {
```

Comment:
Each `document` should have its own `pretranslate` flag, local to the view that is managing this `document`.  I don't think it belongs here in runtime.